app/code/community/Bolt/Boltpay/Helper/ConfigTrait.php

```diff
@@ -120,6 +120,18 @@ public function getPaymentBoltpayConfig($configPath, $checkoutType)
         return ($checkoutType === Bolt_Boltpay_Block_Checkout_Boltpay::CHECKOUT_TYPE_ADMIN) && !Mage::getStoreConfig('payment/boltpay/use_javascript_in_admin') ? '' : $configValue;
     }
 
+    /**
+     * @return array
+     */
+    public function getAllowedButtonByCustomRoutes()
+    {
+        $checkoutType = Bolt_Boltpay_Block_Checkout_Boltpay::CHECKOUT_TYPE_MULTI_PAGE;
+        $config = $this->getPaymentBoltpayConfig('allowed_button_by_custom_routes', $checkoutType);
+        $result = explode(',', $config);
+
+        return (empty($result)) ? [] : $result;
```

Comment:
$result of the explode is never empty
In case of the empty $config string it is
array (
  0 => '',
)
which does not brake things in this particular case but is incorrect.
you can first filter out empty values and then check if it is empty
array_filter(explode(',', $config));

Reply:
thanks, updated.